src/box/memtx_engine.c

```diff
@@ -178,12 +178,16 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 	int rc;
 	struct xrow_header row;
 	uint64_t row_count = 0;
+	int is_space_system = -1;
 	while ((rc = xlog_cursor_next(&cursor, &row,
-				      memtx->force_recovery)) == 0) {
+		/* In case when we read system space, we can't ignore errors */
```

Comment:
Basically, you can factor out first xlog_cursor_next() call since it is the only case when we substitute memtx->force_recovery with false (in all next calls to xlog_cursor_next() is_space_system will be either false or true).
I've reverted this change and all tests seem to pass. Could you add test covering this case?
Also we do not leave comments between arguments and it is not preferable to evaluate non-trivial condition in-place of argument.